### ISSUE TITLE
Problems about showing

### ISSUE THREAD

**Issue by sweetdoggy** [JakobRat/RALF#4]
Dear Jakob

I've successfully run the project on IIC-OSCI-TOOLS,and the RL_placement is done,I have got the DiffAmp_placement pkl file.However, when I try to open this pkl file in magic tools,as you say run python3 main_place_circuit py command(step 3 in your guide).It shows in  MobaXterm that the current technology is "minimum"(as shown in fig1).I'm not sure if there are some problems with the current technology path settings or somewhat.And then I run the step 4 to do a routing.It successfully generate the DiffAmp_routing.tcl file(as shown in fig2).then I run the magic Magic/Placement/<CIRCUIT_NAME>.mag,the placement result is shown in MobaXterm(as shown in fig3),but I don't know how to show all the layers.Then I run the next command source Magic/Routing/DiffAmp_routing.tcl.It shows that the routing result cannot found(as shown in fig4),but I can find the wire connection information in DiffAmp_routing.tcl file(as shown in fig5)

I think the placement and routing steps have been successfully executed,because the corresponding files are successfully generated.But I can not show them with magic tools in MobaXterm.Could you give me some guidance to correctly show the placement and routing result. 

Regards.
Jingyi

Contact: 940819242zjy@gmail.com
![fig1](https://github.com/user-attachments/assets/167d10c4-79bf-4cce-84f1-1966aec139c9)
![fig2](https://github.com/user-attachments/assets/2b4221b3-490d-49c6-b6fc-0eab4fc9b828)
![fig3](https://github.com/user-attachments/assets/0061ff99-ca8f-424d-b989-994f2809e219)
![fig4](https://github.com/user-attachments/assets/bab2656b-58b5-4b94-872a-d76e871bd51a)
![fig5](https://github.com/user-attachments/assets/83b23fd7-f59f-4541-abb2-946e593102d2)

**Comment by JakobRat** [comment]
Dear Jingyi

Congrats on successfully running the project on IIC-OSIC-TOOLS. 

1. To your first question, make sure you are using the SKY130-PDK. Use the `env`-command and check the environmental variable `PDK=sky130A`. If it is not correctly set, you can set it by `export PDK=sky130A`. (For further details about the IIC-OSIC-TOOLS please check the [README](https://github.com/iic-jku/IIC-OSIC-TOOLS/blob/main/README.md).)

2. To show all layers, you can use the `expand` command in Magic (Fig 1). First, select the top cell (`select top cell`) and then expand the cell (`expand`). Make sure you use the terminal which pops up with Magic. [Here](http://opencircuitdesign.com/magic/), you can find more details about commands in Magic.

3. To do the routing in Magic you have to use the terminal which opens with Magic. In that terminal run `source Magic/Routing/DiffAmp_routing.tcl`, as in Fig 2. Alternatively, you can also use the [`main_place_route_circuit.py`](https://github.com/JakobRat/RALF/blob/main/main_place_route_circuit.py) script to show the placement and routing in Magic. For that, make sure you have successfully done the placement (`main_RL_placement.py` or `main_RP_placement.py`) and routing (`main_routing.py`), and correctly set the `CIRCUIT_NAME` variable in  `main_place_route_circuit.py`. 

So in summary, first check the PDK environmental variable, then do the placement (`main_RL_placement.py` or `main_RP_placement.py`), view, and place it in Magic by `main_place_circuit.py`. Then do the routing (`main_routing.py`) and view it in Magic by entering `source Magic/Routing/<CIRCUIT_NAME>_routing.tcl` into the terminal, which opens with Magic. 

I hope I have answered all of your questions. Feel free to ask if anything further is unclear. 

I wish you all the best and much joy with RALF,
Jakob

![image](https://github.com/user-attachments/assets/51b25d33-3016-4d21-9fb2-b20c13d60699)
![image](https://github.com/user-attachments/assets/783a6333-92d6-40d0-870e-ba19bed116b3)

**Comment by sweetdoggy** [comment]
Dear Jakob
I successfully show the placement and routing results of the DiffAmp.I really appreciate your help in solving these problems,and I am trying to make some modifications based on your project,such as applying some other RL algorithms like DQN to implement place or routing.
Regards.
Jingyi
